src/meta/meta_table.c

```diff
@@ -345,28 +345,20 @@ __wt_metadata_salvage(WT_SESSION_IMPL *session)
 }
 
 /*
- * __wt_metadata_uri_to_btree_id --
- *     Given a uri, find the btree id from the metadata. WT_NOTFOUND is returned for a non-file uri.
+ * __wt_metadata_verify --
+ *     Verify the metadata file.
  */
 int
-__wt_metadata_uri_to_btree_id(WT_SESSION_IMPL *session, const char *uri, uint32_t *btree_id)
+__wt_metadata_verify(WT_SESSION_IMPL *session)
 {
-    WT_CONFIG_ITEM id;
-    WT_DECL_RET;
-    char *value;
-
-    value = NULL;
-
-    if (!WT_PREFIX_MATCH(uri, "file:"))
-        return (WT_NOTFOUND);
-
-    WT_ERR(__wt_metadata_search(session, uri, &value));
-    WT_ERR(__wt_config_getones(session, value, "id", &id));
-    *btree_id = (uint32_t)id.val;
+    WT_SESSION *wt_session;
 
-err:
-    __wt_free(session, value);
-    return (ret);
+    wt_session = &session->iface;
+    /*
+     * Now verify the metadata. We know we're in wiredtiger_open and single threaded.
+     */
+    WT_RET(wt_session->verify(wt_session, WT_METAFILE_URI, NULL));
+    return (0);
```

Comment:
I wouldn't bother having a separate function here given we know we're in `wiredtiger_open()`, but I wouldn't fight about it either.

Reply:
The only reason I put it there was to be co-located with `wt_metadata_salvage`. But that could also go away too.

Reply:
Fixed.